libs/sso-client/src/lib/sso-client.ts

```diff
@@ -93,14 +93,24 @@ export const callbackApiHandler =
     }
   };
 
+const validCallbacks = [process.env.NEXT_PUBLIC_SSO_MOCK_APP_URL];
+
 /**
  * Calls the app's callback API route which sets the token as a cookie on the app
  *
  * @param callback Callback URL for the app
  * @param token Access token obtained from login
- * @returns object containing `data` property
+ * @returns object containing `redirect` property or `null` if callback URL is not allowed
  */
 export async function ssoCallback(callback: string, token: string) {
+  if (
+    !validCallbacks
+      .map((it) => it.split('/', 1)[0])
+      .includes(callback.split('/', 1)[0])
```

Comment:
this compares if the protocol part of the input `callback` URL match one of the ones from `validCallbacks`? thought it would be just checking if `callback` is in the allow-list though?

Reply:
Oops yea I don't know what I was thinking with this one
Done, changed to extract the host using the standard URL API lol